src/DynamicExpressions.jl

```diff
@@ -4,6 +4,7 @@ using DispatchDoctor: @stable, @unstable
 
 @stable default_mode = "disable" begin
     include("Utils.jl")
+    include("TypeInterface.jl")
```

Comment:
Might want to have

```julia
using TypeInterfaceModule: is_valid, is_valid_array
```

at the top level, then one can just do

```julia
DynamicExpressions.is_valid(::MyType) = ...
```

to define a custom behavior, without needing to write out the submodule.